src/main/java/com/powsybl/openloadflow/network/impl/PropagatedContingency.java

```diff
@@ -313,12 +314,12 @@ public Optional<LfContingency> toLfContingency(LfNetwork network, boolean useSma
         // reset connectivity to discard triggered branches
         connectivity.reset();
 
-        Map<LfShunt, Double> shunts = new HashMap<>(1);
+        Map<LfShunt, AdmittanceShift> shunts = new HashMap<>(1);
         for (var e : shuntIdsToShift.entrySet()) {
             LfShunt shunt = network.getShuntById(e.getKey());
             if (shunt != null) { // could be in another component
-                double oldB = shunts.getOrDefault(shunt, 0d);
```

Comment:
@geofjamg I have just a doubt on these lines: are we sure that it is equivalent?

Reply:
it is equivalent because AdmittanceShift() init b and g to zero